src/backend/utils/fmgr/fmgr.c

```diff
@@ -776,32 +784,24 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 
 	if (set_sql_dialect && IsTransactionState())
 	{
-		HeapTuple	tuple;
-		Form_pg_proc procedureStruct;
-		tuple = SearchSysCache1(PROCOID,
-								ObjectIdGetDatum(fcinfo->flinfo->fn_oid));
-
-		if (!HeapTupleIsValid(tuple))
-			elog(ERROR, "cache lookup failed for function %u",
-				 fcinfo->flinfo->fn_oid);
-		procedureStruct = (Form_pg_proc) GETSTRUCT(tuple);
-
-		if ((procedureStruct->prolang == pltsql_lang_oid) || (procedureStruct->prolang == pltsql_validator_oid))
+		if ((fcache->prolang == pltsql_lang_oid) || (fcache->prolang == pltsql_validator_oid))
 			sql_dialect_value = tsql_dialect;
 		else
 		{
 			/* Record PG procedure entry */
-			Oid sys_nspoid = get_namespace_oid("sys", false);
+			if (fcache->sys_nspoid == InvalidOid)
```

Comment:
We may need to initialize `sys_nspoid` to `InvalidOid`  in `fmgr_security_definer`.